Modules/Core/Doctrine/HasDynamicRelationships.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Modules\Core\Doctrine;
+
+trait HasDynamicRelationships
+{
+    /**
+    * Dynamically retrieve attributes on the model.
+    *
+    * @param string $key
+    * @return mixed
+    */
+    public function __get($key)
+    {
+        if ($this->setDynamicRelation($key)) {
+            return $this->getRelation($key)->get();
+        }
+
+        return parent::__get($key);
+    }
+
+    /**
+    * Handle dynamic method calls into the model.
+    *
+    * @param string $method
+    * @param array $parameters
+    * @return mixed
+    */
+    public function __call($method, $parameters)
+    {
+        if ($this->setDynamicRelation($method)) {
+            return $this->getRelation($method);
+        }
+
+        return parent::__call($method, $parameters);
+    }
+
+    /**
+     * Set dynamic relationship to the model.
+     *
+     * @param string $key
+     * @return bool
+     */
+    protected function setDynamicRelation($key)
+    {
+        $config = "asgard.user.config.relations.{$key}";
```

Comment:
How come this is namespaced to the `user` module config?

Reply:
Ahh, my bad... i was testing on user module and totally forget to make that dynamic.

Reply:
Thanks for pointing that out